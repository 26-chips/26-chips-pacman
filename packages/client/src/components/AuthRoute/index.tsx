import type { ReactNode } from 'react';
import { Navigate } from 'react-router-dom';
import { ROUTES } from 'router';
import { useFetchUserQuery } from 'api';
import { Loader } from 'components';

type AuthRouteProps = {
  children: ReactNode;
};

export const AuthRoute = ({ children }: AuthRouteProps): JSX.Element => {
  const { data: user, isLoading } = useFetchUserQuery();

  return isLoading ? (
    <Loader />
  ) : user ? (
    <Navigate to={ROUTES.START} />
  ) : (
    <>{children}</>
  );
};
