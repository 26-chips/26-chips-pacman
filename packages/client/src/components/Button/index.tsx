import { ButtonHTMLAttributes, ReactNode } from 'react';
import cn from 'classnames';
import styles from './styles.module.scss';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  thema?: 'default' | 'light';
  width?: number;
  height?: number;
  loading?: boolean;
}

export const Button = (props: ButtonProps) => {
  const {
    thema = 'default',
    width,
    height,
    children,
    loading,
    ...restProps
  } = props;

  return (
    <button
      className={cn(styles.button, {
        [styles.light]: thema === 'light',
        [styles.loading]: loading,
      })}
      style={{
        width: width && `${width}px`,
        height: height && `${height}px`,
      }}
      {...restProps}>
      {loading ? 'Loading...' : children}
    </button>
  );
};
