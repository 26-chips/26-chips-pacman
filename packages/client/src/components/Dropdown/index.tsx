import { ReactNode, useCallback, useEffect, useRef, useState } from 'react';
import styles from './styles.module.scss';
import ArrowDown from 'assets/icons/arrow-down.svg';
import cn from 'classnames';

interface IProps {
  title: string;
  variants: ReactNode[];
  defaultActiveItem?: number;
  onSelect?: (label: string) => void;
  defaultPosition?: 'left' | 'right';
  hideAfterSelect?: boolean;
}

export const Dropdown = ({
  title,
  variants,
  defaultActiveItem,
  defaultPosition = 'left',
  hideAfterSelect = true,
}: IProps) => {
  const dropdownWidth = 262;
  const [showList, setShowList] = useState(false);
  const [position, setPosition] = useState(defaultPosition);
  const [activeItem, setActiveItem] = useState(defaultActiveItem);
  const dropdownRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    const { current } = dropdownRef;
    if (!current) return;
    if (current.offsetLeft <= dropdownWidth) setPosition('right');
    else if (current.offsetLeft + dropdownWidth >= window.innerWidth)
      setPosition('left');
  }, []);

  const onClickEvent = useCallback((e: MouseEvent) => {
    if (
      e.target === dropdownRef.current ||
      dropdownRef.current?.contains(e.target as Node)
    ) {
      return;
    }
    setShowList(false);
    window.removeEventListener('click', onClickEvent);
  }, []);

  const handleClick = () => {
    if (showList) {
      window.removeEventListener('click', onClickEvent);
    } else {
      window.addEventListener('click', onClickEvent);
    }
    setShowList(!showList);
  };

  const handleSelect = (index: number) => {
    setActiveItem(index);
    if (hideAfterSelect) {
      setShowList(false);
      window.removeEventListener('click', onClickEvent);
    }
  };

  return (
    <div className={styles.dropdown} ref={dropdownRef}>
      <span className={styles.title} onClick={handleClick}>
        {title}
        <img
          className={cn({ [styles.isUp]: showList })}
          src={ArrowDown}
          alt="arrow-down"
        />
      </span>
      <div
        className={cn(styles.menu, styles[position], {
          [styles.isOpen]: showList,
        })}>
        <ul className={styles.list}>
          {variants.map((elem, index) => (
            <div
              key={index}
              className={cn({ [styles.isActive]: index === activeItem })}
              onClick={() => handleSelect(index)}>
              {elem}
            </div>
          ))}
        </ul>
      </div>
    </div>
  );
};
