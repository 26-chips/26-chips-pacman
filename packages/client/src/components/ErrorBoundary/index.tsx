import { ErrorPage } from 'pages';
import { useRouteError, isRouteErrorResponse } from 'react-router-dom';

export const ErrorBoundary = () => {
  const error = useRouteError();

  const internalErrorProps = {
    statusCode: '500',
    errorText: 'Внутренняя ошибка сервера :(',
    image: 'Blue',
  };

  if (isRouteErrorResponse(error)) {
    return <ErrorPage />;
  } else {
    return (
      //@ts-ignore
      <ErrorPage {...internalErrorProps} />
    );
  }
};

// type ParentProps = {
//     children: ReactNode;
// };
//
// type State = {
//     error: boolean;
//     errorInfo: string;
// };
//
// export class ErrorBoundary extends Component {
//     constructor(props: ParentProps) {
//         super(props);
//         this.state = { error: false, errorInfo: '' };
//     }
//
//     componentDidCatch(error: Error) {
//         console.error(error)
//
//         this.setState({
//             error: Boolean(error),
//             errorInfo: error.message
//         })
//     }
//
//     render() {
//         if (this.state.errorInfo) {
//             return
//             <ErrorPage
//                 statusCode = '500'
//                 errorText = 'Внутренняя ошибка сервера :(\nМы о ней знаем и скоро все исправим!'
//                 image = 'Blue'
//             />;
//         }
//         return this.props.children;
//     }
// }
