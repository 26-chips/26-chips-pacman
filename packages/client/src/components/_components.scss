@import "../constants";

.helper {
  display: none;
  position: absolute;
  font-size: 0.9rem;
  margin: 0.1rem 0.8rem;
  letter-spacing: 0.02rem;
  color: $color-font;
}

.danger {
  textarea {
    border-color: $color-warning;

    &:focus {
      outline: none;
      box-shadow: inset 0 0.05rem 0.1rem 0.01rem $color-warning;
    }
  }

  input {
    border-color: $color-warning;
  }

  .helper {
    display: block;
  }

  .helper, .input-label {
    color: $color-warning;
  }
}

.hide {
  display: none;
}
