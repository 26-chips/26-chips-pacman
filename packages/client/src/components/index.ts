export * from './Menu';
export * from './Input';
export * from './Button';
export * from './Checkbox';
export * from './Tabs';
export * from './Modal';
export * from './Switch';
