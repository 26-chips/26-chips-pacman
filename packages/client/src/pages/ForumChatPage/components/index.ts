export * from './participantsList';
export * from './messageLayout';
export * from './emojisList';
export * from './addEmojiButton';
export * from './emojiButton';
export * from './participantItem';
