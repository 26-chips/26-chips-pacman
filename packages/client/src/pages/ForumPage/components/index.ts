export * from './createTopic';
export * from './forumTopic';
export * from './forumTopicsTable';
