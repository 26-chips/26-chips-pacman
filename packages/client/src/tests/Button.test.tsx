import { Button } from 'components';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';

const mockComponent = <Button>Default Button</Button>;

describe('<Button />', () => {
  it('Should render correctly', () => {
    const button = render(mockComponent);
    expect(button.getByText('Default Button')).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const tree = create(mockComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
