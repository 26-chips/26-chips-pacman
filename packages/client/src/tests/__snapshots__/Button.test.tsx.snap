// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should match snapshot 1`] = `
<button
  className="undefined undefined undefined"
  type="button"
>
  Default Button
</button>
`;
